Configuration/Skimming/src/LeptonSkimming.cc

```diff
@@ -0,0 +1,487 @@
+ PtMu_Cut=runParameters.getParameter<double>("PtMu_Cut");
+  PtEl_Cut=runParameters.getParameter<double>("PtEl_Cut");
+ QualMu_Cut=runParameters.getParameter<double>("QualMu_Cut");
+ MuTrgExclusionCone=runParameters.getParameter<double>("MuTrgExclusionCone");
+ ElTrgExclusionCone=runParameters.getParameter<double>("ElTrgExclusionCone");
+ TrkObjExclusionCone=runParameters.getParameter<double>("TrkObjExclusionCone");
+ MuTrgMuDz_Cut=runParameters.getParameter<double>("MuTrgMuDz_Cut");
+ ElTrgMuDz_Cut=runParameters.getParameter<double>("ElTrgMuDz_Cut");
+ ObjPtLargerThanTrack=runParameters.getParameter<bool>("ObjPtLargerThanTrack");
+
+
+
+}
+
+
+LeptonSkimming::~LeptonSkimming()
+{
+ 
+   // do anything here that needs to be done at destruction time
+   // (e.g. close files, deallocate resources etc.)
+
+}
+
+bool LeptonSkimming::sort_by_MaxPt(const std::vector<float>& a, const std::vector<float>& b) {
+  return a[0] > b[0];
+  }
+
+bool LeptonSkimming::HLTFired(const edm::Event& iEvent, const edm::EventSetup& iSetup,std::vector<string> HLTPath ){
+  using namespace std;  using namespace edm;  using namespace reco;
+  using namespace trigger;
+ 
+  edm::Handle<edm::TriggerResults> trigResults;
+  iEvent.getByToken(trgresultsToken_, trigResults);
+ 
+  
+  bool fire= false;
+  if( trigResults.failedToGet() ) return false;
+  for (unsigned int ip=0; ip<HLTPath.size(); ip++){
+  int N_Triggers = trigResults->size();
+  const edm::TriggerNames & trigName = iEvent.triggerNames(*trigResults);    
+  //cout << "new" << endl;
+  for( int i_Trig = 0; i_Trig < N_Triggers; ++i_Trig ) {
+    std::string TrigPath =trigName.triggerName(i_Trig); 
+    if (!trigResults->accept(i_Trig))   continue;
+    if (TrigPath.find(HLTPath[ip]) != std::string::npos) fire=true;
+    } 
+  }
+  return fire;
+}
+
+std::vector<float> LeptonSkimming::HLTObject(const edm::Event& iEvent, const edm::EventSetup& iSetup,std::vector<string> Seed ){
+  using namespace std;  using namespace edm;  using namespace reco;
+  using namespace trigger;
+ 
+  edm::Handle<trigger::TriggerEvent> triggerObjectsSummary;
+  iEvent.getByToken(trigobjectsToken_ ,triggerObjectsSummary);
+  trigger::TriggerObjectCollection selectedObjects;
+
+  std::vector<std::vector<float>> max_per_trigger; 
+
+for (unsigned int ipath=0; ipath<Seed.size(); ipath++){ 
+  std::vector<std::vector<float>> tot_tr_obj_pt_eta_phi;
+  if (!triggerObjectsSummary.isValid()) continue;
+    size_t filterIndex = (*triggerObjectsSummary).filterIndex(InputTag(Seed[ipath],"","HLT"));
+    trigger::TriggerObjectCollection allTriggerObjects = triggerObjectsSummary->getObjects();     
+    if (filterIndex < (*triggerObjectsSummary).sizeFilters()) { 
+      const trigger::Keys &keys = (*triggerObjectsSummary).filterKeys(filterIndex);
+      for (size_t j = 0; j < keys.size(); j++) {
+        trigger::TriggerObject foundObject = (allTriggerObjects)[keys[j]];
+        std::vector<float> tr_obj_pt_eta_phi;
```

Comment:
better as a std:array<float,5> (which also avoids the vector of vectors above)

Reply:
changed. Now we have array 5